### PR TITLE
Add navigation messages with persist logs enabled.

### LINKED ISSUE #68: Add support for navigation marker
Testing: 
    - Set `devtools.webconsole.persistlog` to true in about:config
    - Navigate to a different page
[HTML and packets](https://github.com/linclark/webconsole-rdp/tree/master/data/tabNavigated)
[Integration point](https://dxr.mozilla.org/mozilla-central/source/devtools/client/webconsole/webconsole.js#1984)

### PR DESCRIPTION
**Pull request by pelmers**
Fixes #68.

I still need to write some test, but I put this up for any UI feedback. Perhaps that pends #181 though.
screen shot 2016-07-29 at 11 17 49 am

### CODE REVIEW

File: devtools/client/webconsole/new-console-output/utils/messages.js [hunk 1]
```diff
@@ -85,6 +86,17 @@ function transformPacket(packet) {
       });
     }
 
+    case "navigationMessage": {
+      let { message } = packet;
+      return new ConsoleMessage({
+        source: MESSAGE_SOURCE.CONSOLE_API,
+        type: MESSAGE_TYPE.CONSOLE_API,
```
Comment (linclark) [inline devtools/client/webconsole/new-console-output/utils/messages.js]: There isn't a MESSAGE_TYPE.CONSOLE_API. There is a MESSAGE_TYPE.LOG, which would probably be appropriate here.

**Comment by linclark** [comment]
I think this looks good for now. We'll probably have to put more thought into all of the system messages holistically for #181, but this will help users in the meantime.

**Comment by pelmers** [comment]
@bgrins @linclark hi, I pushed a work-in-progress on writing a test for the navigation marker. I'm not sure why it fails the mocha test (the actual and expected objects it prints out look very different...).

But I'm away for the next 3 weeks and wouldn't be able to finish it until returning; I figured I'd put up what I have in case you would like to pick it up.

**Comment by linclark** [comment]
Thanks for posting this! I pulled in some of the changes in #215. I'm going to postpone the test because we're switching around how stubs are generated.

**Pull request closed without merging**